extensions/ql-vscode/src/mocks/vscode-mock-gh-api-server.ts

```diff
@@ -27,7 +27,10 @@ export class VSCodeMockGitHubApiServer extends DisposableObject {
   private readonly server: MockGitHubApiServer;
   private readonly config: MockGitHubApiConfigListener;
 
-  constructor(private readonly ctx: ExtensionContext) {
+  constructor(
+    private readonly ctx: ExtensionContext,
+    private readonly app: App,
```

Comment:
It feels strange to be passing in both the context and the app. I believe both of the properties that this class accesses on the context (`extensionMode` and `extensionUri`) also exist on the app as `mode` and `extensionPath`. I think it would be best to switch to those in this PR as well, even though the change is unrelated.

Reply:
That sounds good to me. I didn't really have it clear in my head what the point of `ExtensionContext` was and how it relates to the app, but I see now the app can replace the extension context in some places. The context is stored as a field on the app, so it can act as a platform-independent interface to those values.

I can take a look (not in this PR) to see if there are more places where we pass both types and if it could be simplified.